### PR TITLE
fix : updating ilens with api

### LINKED ISSUE #149: Ilens - Which Column to Show to Green
In Ilens the data column is mismatching with response from the api. Hence need a confirm from customer which data is actually needed to be shown in Green

image

### PR DESCRIPTION
**Pull request by anagarwa**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #149 

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://ilensapi--icicidirect--aemsites.hlx.live/research/equity

### CODE REVIEW

File: blocks/i-lens/i-lens.js [hunk 1]
```diff
@@ -114,26 +87,77 @@ function addStocksData(ilensBody, dropdown = 'default') {
...
+  // console.log(dropdownValue);
+  // const menuItems = dropdownValue.split(', ');
+  const dropdownText = menuItems[0].text;
+
+  const dropdownSelectDiv = document.createElement('div');
+  dropdownSelectDiv.className = 'dropdown-select';
+
+  const button = document.createElement('button');
+  button.className = 'dropdown-toggle';
+  button.innerHTML = `<span class="dropdown-text">${dropdownText}</span><span class="icon-down-arrow icon"></span>`;
```
Comment (ravverma) [inline blocks/i-lens/i-lens.js:100]: just to confirm that the value of `dropdownText ` is coming from json not from the page itself.

File: blocks/i-lens/i-lens.js [hunk 2]
```diff
@@ -114,26 +87,77 @@ function addStocksData(ilensBody, dropdown = 'default') {
...
+    if (item.value) {
+      li.dataset.value = item.value;
+    }
+    const a = document.createElement('a');
+    const span = document.createElement('span');
+    span.textContent = item.text;
+    a.appendChild(span);
+    li.appendChild(a);
+    li.addEventListener('click', (event) => {
+      // eslint-disable-next-line no-use-before-define
```
Comment (ravverma) [inline blocks/i-lens/i-lens.js:119]: can get rid of this by placing `updateRecommedations ` above from this.

File: blocks/i-lens/i-lens.js [hunk 3]
```diff
@@ -114,26 +87,77 @@ function addStocksData(ilensBody, dropdown = 'default') {
...
-      if (item.primary === dropdownValue) {
-        const secondDropDownValue = JSON.parse(item.secondary).join(', ');
-        const secondDropDown = createDropdown(secondDropDownValue);
-        dropdownsDiv.appendChild(secondDropDown);
-        addStocksData(ilensBody, dropdownValue.toLowerCase());
-      }
-    });
+  const secondaryValues = [];
+  dropDownDetails.forEach((item) => {
+    // console.log(item);
```
Comment (ravverma) [inline blocks/i-lens/i-lens.js:142]: console

File: blocks/i-lens/i-lens.js [hunk 4]
```diff
@@ -145,28 +169,35 @@ function closeAllDropDowns(clickedElement) {
...
-  const header = document.createElement('div');
-  header.className = 'i-lens-header';
-  const rowDiv = document.createElement('div');
-  rowDiv.className = 'row align-items-center';
+function updateDropDownList(ilensContainer, dropDownData) {
+  const rowDiv = ilensContainer.querySelector('.row');
+  const dropDownDetails = [];
+  const primaryDropDown = [];
+  dropDownData.forEach((item) => {
+    // console.log(item);
```
Comment (ravverma) [inline blocks/i-lens/i-lens.js:177]: console

**Review by ravverma — CHANGES_REQUESTED** [review]

File: blocks/i-lens/i-lens.js (hunk 1 above)
Comment (anagarwa) [reply blocks/i-lens/i-lens.js:100]: @ravverma yes as confirmed from here https://github.com/aemsites/icicidirect/pull/205/files#diff-1d24df7cd7f8c1428271be3a80d890c3ba785bee6dededd50cdb60a2b98adfb4R93

File: blocks/i-lens/i-lens.js (hunk 2 above)
Comment (anagarwa) [reply blocks/i-lens/i-lens.js:119]: it is sort of creating a cycle, have to keep this comment at one of the method

File: blocks/i-lens/i-lens.js (hunk 3 above)
Comment (anagarwa) [reply blocks/i-lens/i-lens.js]: fixed

File: blocks/i-lens/i-lens.js (hunk 4 above)
Comment (anagarwa) [reply blocks/i-lens/i-lens.js]: fixed

**Pull request merged**